packages/esm-patient-chart-app/src/visit/visit-form/visit-form.component.tsx

```diff
@@ -219,87 +279,12 @@ const StartVisitForm: React.FC<StartVisitFormProps> = ({
     formState: { errors, isDirty },
-    setError,
   } = methods;
 
   useEffect(() => {
     promptBeforeClosing(() => isDirty);
   }, [isDirty, promptBeforeClosing]);
 
-  let [maxVisitStartDatetime, minVisitStopDatetime] = useMemo(() => {
-    if (!visitToEdit?.encounters?.length) {
-      return [null, null];
-    }
-
-    const allEncountersDateTime = visitToEdit?.encounters?.map(({ encounterDatetime }) =>
-      Date.parse(encounterDatetime),
-    );
-    const maxVisitStartDatetime = Math.min(...allEncountersDateTime);
-    const minVisitStopDatetime = Math.max(...allEncountersDateTime);
-    return [maxVisitStartDatetime, minVisitStopDatetime];
-  }, [visitToEdit]);
-
-  const validateVisitStartStopDatetime = useCallback(() => {
-    let visitStartDate = getValues('visitStartDate');
-    const visitStartTime = getValues('visitStartTime');
-    const visitStartTimeFormat = getValues('visitStartTimeFormat');
-
-    const [visitStartHours, visitStartMinutes] = convertTime12to24(visitStartTime, visitStartTimeFormat);
-
-    const visitStartDatetime = visitStartDate.setHours(visitStartHours, visitStartMinutes);
-
-    let validSubmission = true;
-
-    if (maxVisitStartDatetime && visitStartDatetime >= maxVisitStartDatetime) {
-      validSubmission = false;
-      setError('visitStartDate', {
-        message: t('invalidVisitStartDate', 'Start date needs to be on or before {{firstEncounterDatetime}}', {
-          firstEncounterDatetime: new Date(maxVisitStartDatetime).toLocaleString(),
-          interpolation: {
-            escapeValue: false,
-          },
-        }),
-      });
-    }
-
-    if (!displayVisitStopDateTimeFields) {
-      return validSubmission;
-    }
-
-    let visitStopDate = getValues('visitStopDate');
-    const visitStopTime = getValues('visitStopTime');
-    const visitStopTimeFormat = getValues('visitStopTimeFormat');
-
-    const [visitStopHours, visitStopMinutes] = convertTime12to24(visitStopTime, visitStopTimeFormat);
-
-    const visitStopDatetime = visitStopDate.setHours(visitStopHours, visitStopMinutes);
-
-    if (minVisitStopDatetime && visitStopDatetime <= minVisitStopDatetime) {
-      validSubmission = false;
-      setError('visitStopDate', {
-        message: t(
-          'visitStopDateMustBeAfterMostRecentEncounter',
-          'Stop date needs to be on or after {{lastEncounterDatetime}}',
-          {
-            lastEncounterDatetime: new Date(minVisitStopDatetime).toLocaleString(),
-            interpolation: {
-              escapeValue: false,
-            },
-          },
-        ),
-      });
-    }
-
-    if (visitStartDatetime >= visitStopDatetime) {
-      validSubmission = false;
-      setError('visitStopDate', {
-        message: t('invalidVisitStopDate', 'Visit stop date time cannot be on or before visit start date time'),
-      });
-    }
-
-    return validSubmission;
-  }, [displayVisitStopDateTimeFields, getValues, maxVisitStartDatetime, minVisitStopDatetime, setError, t]);
```

Comment:
I've moved all of this logic up into the Zod schema where it's hopefully easier to reason about. Other benefits to this maybe that:

- The validation logic is centralised in one place
- The validation automatically runs on form interactions
- The logic can still be extended to handle more complex validations with refinements to the schema